Lessons_18/task1_with_reg_ex.py

```diff
@@ -0,0 +1,63 @@
+"""Task 1
+Create a class method named 'validate', which should be called from the 
+'__init__' method to validate parameter email, passed to the constructor.
+The logic inside the 'validate' method could be to check if the passed email
+parameter is a valid email string.
+"""
+
+import re
+
+
+class User:
+    """Create user object"""
+
+    def __init__(self, login, name, email):
+        self.login = login
+        self.name = name
+        self.email = User.validate(email) or "Wrong email format!"
+
+    def talk(self):
+        """Print user greeting"""
+        print(
+            f"Hello, I am {self.name}! My user login {self.login}.\n"
+            f"You can send me email: {self.email}"
+        )
+
+    @classmethod
+    def validate(cls, email):
+        """Check if the passed email parameter is a valid email string"""
+        if isinstance(email, str):
+            email_regex_string = (
+                r"([A-Za-z0-9]+[-._])*[A-Za-z0-9]+@[A-Za-z0-9-]+(\.[A-Z|a-z]{2,})+"
```

Comment:
Would match "user@mail.com", but wouldn't match "user@mail.com.ua" (this part: `{2,}` seems wrong to me)